### PR TITLE
Change this from `noarch: python` to yes-arch

### PR DESCRIPTION
**Pull request by pkgw**
This package includes a binary module, so it can't be `noarch: python`.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by pkgw** [comment]
OK, I've also addressed the linter complaint about `pypi.io`.

For context, I was trying to create recipes for some Python packages that depend on beancount, and they weren't working because they were building with a different minor version of Python, which broke due to the use of the binary extension module. In general, as it currently stands this package is *only* going to work on x86_86 Linux with Python 3.12, so it's a pretty major issue. Fortunately it's not too hard to fix up though.

**Comment by pkgw** [comment]
@kklein If you have a chance to review this pull request, that would be great. If not, I do have permissions to merge it myself as a member of the core team. Whenever possible we do want at least one other person to review any pull requests before merging them, but I'm confident that this PR adds an important fix to this feedstock so I might go ahead and merge it myself if I don't hear anything in the next few days or so.

**Review by kklein — APPROVED** [review]
Hi @pkgw ! Apologies about the delay; I've been sick for a couple of days.
Thanks a lot for taking care of this and fixing it. :)

**Pull request merged**
